src/main/java/com/minecraftabnormals/abnormals_core/core/util/DataUtil.java

```diff
@@ -134,4 +145,101 @@ public static void addToJigsawPattern(ResourceLocation toAdd, JigsawPiece newPie
+	 *
+	 * <p>For example, in a config condition created under the id {@code abnormals_core}
+	 * that checks whether {@code "sign_editing_requires_empty_hand"} (the annotated value for the
+	 * {@code signEditingRequiresEmptyHand} field) is true, the syntax would be like this:</p>
+	 *
+	 * <pre>{@code
+	 * "conditions": [
+	 *   {
+	 *     "type": "abnormals_core:config"
+	 *     "value": "sign_editing_requires_empty_hand"
+	 *   }
+	 * ]
+	 * }</pre>
+	 *
+	 * <p>Config conditions also accept a {@code predicates} array which defines predicates the config value must match
+	 * before the condition returns true, and a boolean {@code inverted} argument which makes the condition pass if it
+	 * evaluates to false instead of true. If the config value is non-boolean, {@code predicates} is required.
+	 * Each individual predicate also accepts an {@code inverted} argument, as {@code !(A.B) != !A.!B}.</p>
+	 *
+	 * <p>For example, you could check whether a the float config value {@code "potato_poison_chance"} is less than
+	 * 0.1 by using the {@code "abnormals_core:greater_than_or_equal_to"} predicate and inverting it. (Of course,
+	 * in this situation it's easier to just use the {@code "abnormals_core:less_than"} predicate, but this is just
+	 * an example used to show the syntax of inverting).</p>
+	 *
+	 * <pre>{@code
+	 * "conditions": [
+	 *   {
+	 *     "type": "abnormals_core:config",
+	 *     "value": "potato_poison_chance",
+	 *     "predicates": [
+	 *       {
+	 *         "type": "abnormals_core:greater_than_or_equal_to",
+	 *         "value": 0.1,
+	 *         "inverted": true
+	 *       }
+	 *     ]
+	 *   }
+	 * ],
+	 * }</pre>
+	 *
+	 * <p>Abnormals Core has pre-made predicates for numeric and string comparison as well as checking for equality,
+	 * but you can create custom predicates and register them with
+	 * {@link DataUtil#registerConfigPredicate(IConfigPredicateSerializer)}.</p>
+	 *
+	 * @param modId The mod ID to register the config condition under. The reason this is required and that you can't just
+	 *              register your values under {@code "abnormals_core:config"} is because there could be duplicate keys
+	 *              between mods.
+	 * @param configObjects The list of objects to get config keys from. The {@link ConfigKey} values must be unique.
+	 *
+	 * @author abigailfails
+	 */
+	public static void registerConfigCondition(String modId, Object... configObjects) {
+		HashMap<String, ForgeConfigSpec.ConfigValue<?>> configValues = new HashMap<>();
+		Arrays.asList(configObjects).forEach(cfg -> Arrays.stream(cfg.getClass().getDeclaredFields()).filter(f -> f.getAnnotation(ConfigKey.class) != null && ForgeConfigSpec.ConfigValue.class.isAssignableFrom(f.getType())).forEach(f -> {
+			f.setAccessible(true);
+			try {
+				configValues.put(f.getAnnotation(ConfigKey.class).value(), (ForgeConfigSpec.ConfigValue<?>) f.get(cfg));
+			} catch (IllegalAccessException ignored) {
+			}
+		}));
```

Comment:
I'm not really a fan of the big use of streams and lambdas here when this could all be done pretty simply without the use of them.
Just iterate over `configObjects`, get the class for each object, and then search for valid fields.

Reply:
Changed to foreach loop in latest commit.